### PR TITLE
Changes to build restapi docker

### PR DESCRIPTION
**Pull request by sumukhatv**
**- What I did**
Make changes to build sonic-restapi docker
https://github.com/Azure/sonic-restapi
**- How I did it**
Add Dockerfiles
Create submodule sonic-restapi to pull code from sonic-restapi repo
**- How to verify it**
should see target/docker-sonic-restapi.gz once built
**- Description for the changelog**

Added supporting files and changes to build sonic-restapi docker

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by lguohan** [comment]
build failure due to permission issue. can you investigate?

https://sonic-jenkins.westus2.cloudapp.azure.com/job/broadcom/job/buildimage-brcm-all-pr/1960/consoleFull

**Comment by sumukhatv** [comment]
Retest broadcom please

**Comment by sumukhatv** [comment]
Retest broadcom please

File: dockers/docker-sonic-restapi/Dockerfile.j2 [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+## Make apt-get non-interactive
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update
+
+## Install redis-tools dependencies
+## TODO: implicitly install dependencies
+RUN apt-get -y install libjemalloc1
+RUN pip install setuptools
```
Comment (lguohan) [inline dockers/docker-sonic-restapi/Dockerfile.j2]: do we need setuptools and libjemalloc1?

**Comment by lguohan** [comment]
in another PR, you need to add critical process like this one. https://github.com/Azure/sonic-buildimage/blob/master/dockers/docker-orchagent/critical_processes

**Pull request merged**